src/core/ddsc/src/dds_rhc_default.c

```diff
@@ -1531,16 +1604,25 @@ static bool dds_rhc_default_store (struct dds_rhc_default * __restrict rhc, cons
          WD     D;gen++     D
          Simplest way is to toggle istate when it is currently DISPOSED
          and the operation is WD. */
-      if (has_data && inst->isdisposed)
+      if (has_data && inst->isdisposed && !is_dispose)
```

Comment:
Why this change? It changes the behaviour of the disposed generations for the case of a write-dispose operation on an already disposed topic. It used to treat as a write followed by a dispose, and hence consider the instance alive, even if that was observable only by the change in the generation; now it simply remains in the disposed state. I might well be inclined to agree with it, just want to know 😄 

(And if the change stands, the comment preceding it needs to be updated)